### DISCUSSION TITLE
Having trouble with Netlify continuous deploy

### DISCUSSION THREAD

**Discussion by zsofiaj** [thegetty/quire#173] [Get Help]
Hello everyone! I'm having trouble getting my Netlify continuous deploy to work for a new Quire project. I've set one up before, and I don't recall having any problems. This time, when I try to deploy the site, the deploy fails. I've pasted the deploy log below. Has anyone dealt with this "refs/heads/main does not exist" issue? I wan't able to find anything in this forum, and I didn't find any answers in other GitHub forums either. Thanks!

8:36:53 AM: Build ready to start
8:36:55 AM: build-image version: fa439ad1ab9393b2c0d449d8d7c033927683f4b0
8:36:55 AM: build-image tag: v4.3.0
8:36:55 AM: buildbot version: 95d3b7a0c6ab391660cf43194e4a771ec6d12c7f
8:36:55 AM: Building without cache
8:36:55 AM: Starting to prepare the repo for build
8:36:55 AM: git ref refs/heads/main does not exist or you do not have permission
8:36:55 AM: Creating deploy upload records
8:36:55 AM: Failing build: Failed to prepare repo
8:36:55 AM: Failed during stage 'preparing repo': git ref refs/heads/main does not exist
8:36:55 AM: Finished processing build request in 871.179891ms

**Comment by Erin-Cecele** [comment]
Hi @zsofiaj. This is not an issue we've seen come up before. Just to check in is your deploy still failing to build? We can look into this for you.

**Reply by zsofiaj** [reply]
Thanks, @Erin-Cecele. I just tried running the build request again, and it again failed to deploy my site. This time I got a much longer deploy log (almost 750 rows). At the very end it says this:

4:50:38 PM: npm ERR! Build failed with error code: 1
4:50:38 PM: npm ERR! A complete log of this run can be found in:
4:50:38 PM: npm ERR!     /opt/buildhome/.npm/_logs/2021-09-17T20_50_38_364Z-debug.log
4:50:38 PM: Error during NPM install
4:50:38 PM: Build was terminated: Build script returned non-zero exit code: 1
4:50:38 PM: Failing build: Failed to build site
4:50:38 PM: Finished processing build request in 3m9.949769316s

**Reply by geealbers** [reply]
Hi, @zsofiaj. Can you try adding the below two lines to your project’s netlify.toml file? This will ensure Netlify is using the LTS version of Node. Hopefully it will help, but let us know. And if you get another failure, please go ahead and share the entire deploy log here. It might help us target some other possible fixes. Thanks!

```toml
[build.environment]
NODE_VERSION = "14.18.0"
```

Here's a screenshot example with a slightly older version of Node:

<img width="711" alt="Screen Shot 2021-09-28 at 10 55 18 AM" src="https://user-images.githubusercontent.com/7796401/135140219-b554db8c-a807-4327-a4fb-022b22ef5139.png">

**Reply by zsofiaj** [reply]
Thanks so much, @geealbers . I tried that, and it didn't seem to work. I just updated Quire, and I'm working on migrating my two existing projects to the new version. I'll try the continuous deploy again once I'm done with that. I have another question about migrating my projects, but I'll start a new thread to ask about that. Thanks again!

**Reply by zsofiaj** [reply]
@geealbers and @Erin-Cecele, I just want to let you both know that once I updated Quire and migrated my two projects over, I was able to set up the Netlify continuous deploy for both. I'm not sure what was wrong earlier. That might remain a mystery. But we could close this thread now since this issue has been resolved. Thank you both for your help!!
